sphinx/ext/intersphinx/_resolve.py

```diff
@@ -64,7 +64,9 @@ def _resolve_reference_in_domain_by_target(
         inv_name: str | None, inventory: Inventory,
         domain: Domain, objtypes: Iterable[str],
         target: str,
-        node: pending_xref, contnode: TextElement) -> nodes.reference | None:
+        node: pending_xref, contnode: TextElement,
+        app: Sphinx | None = None
```

Comment:
This is ungreat, I admit - both in terms of adding a single default argument to an otherwise entirely-positional (and thus strict) interface, and also in terms of polluting the resolution functions by providing them with (potential) access to the `app` object.

If there's a better way (and there probably is) I'll be glad for suggestions about that.